### DISCUSSION TITLE
Prisma VS code extension giving warnings for multi column index

### DISCUSSION THREAD

**Discussion by Aaditya-23** [prisma/prisma#18718] [Q&A] [answered]
### Bug description

If I define @@index on two fields in same line, I get a warning in console to define index.

> With `relationMode = "prisma"`, no foreign keys are used, so relation fields will not benefit from the index usually created by the relational database under the hood. This can lead to poor performance when querying these fields. We recommend adding an index manually. Learn more at https://pris.ly/d/relation-mode-prisma-indexes" 

 
![Model](https://user-images.githubusercontent.com/74983108/230781867-88dd4ba3-f7f8-48ef-a3bb-2e637275d56c.png)

But if I define @@index in two different lines, it works as expected.

![Model](https://user-images.githubusercontent.com/74983108/230781994-3a2c0a82-389e-471c-a9a6-977173c76a2c.png)

I am using planetscale, so I need to add indexes on fk.

I am not sure if we can create multi-column indexes on the same line. Docs show we can, then why do I get this warning?

### How to reproduce

Create a prisma file and add this schema or add another schema with multi-column index to reproduce this warning.  

### Prisma information

```prisma

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider     = "postgresql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

model User {
  id       String    @id @default(uuid())
  posts    Post[]
  comments Comment[]
}

model Post {
  id      String    @id @default(uuid())
  userId  String?
  User    User?     @relation(fields: [userId], references: [id])
  comment Comment[]

  @@index([userId])
}

model Comment {
  id     String  @id @default(uuid())
  User   User?   @relation(fields: [userId], references: [id])
  userId String?
  Post   Post?   @relation(fields: [postId], references: [id])
  postId String?

  @@index([userId, postId])
}

```

 

### Environment & setup

- OS: Windows
- Database: PlanetScale
- Node.js version: 18.14.1

### Prisma Version

```

prisma                  : 4.12.0
@prisma/client          : 4.12.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7 (at node_modules\@prisma\engines\migration-engine-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.12.0-67.659ef412370fa3b41cd7bf6e94587c1dfb7f67e7
Default Engines Hash    : 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7
Studio                  : 0.483.0

```

**Comment by ludralph** [comment, marked as answer]
Hi @Aaditya-23 👋 

When working with Planetscale, you will need to create the index on multiple relation scalar fields on a separate line instead of using multi-column index.

```js
model User {
  id       String    @id @default(uuid())
  posts    Post[]
  comments Comment[]
}

model Post {
  id      String    @id @default(uuid())
  userId  String?
  User    User?     @relation(fields: [userId], references: [id])
  comment Comment[]

  @@index([userId])
}

model Comment {
  id     String  @id @default(uuid())
  User   User?   @relation(fields: [userId], references: [id])
  userId String?
  Post   Post?   @relation(fields: [postId], references: [id])
  postId String?

  @@index([userId])
  @@index([postId])
}
```

**Reply by ludralph** [reply]
Hi @Aaditya-23 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by Aaditya-23** [reply]
thanks @ludralph
